x-pack/plugins/observability_solution/apm/server/routes/entities/services/routes.ts

```diff
@@ -61,91 +52,38 @@ const servicesEntitiesRoute = createApmServerRoute({
 const servicesEntitiesDetailedStatisticsRoute = createApmServerRoute({
   endpoint: 'POST /internal/apm/entities/services/detailed_statistics',
   params: t.type({
-    query: t.intersection([
-      environmentRt,
-      kueryRt,
-      rangeRt,
-      t.intersection([offsetRt, probabilityRt, serviceTransactionDataSourceRt]),
-      t.type({
-        bucketSizeInSeconds: toNumberRt,
-      }),
-    ]),
+    query: t.intersection([environmentRt, kueryRt, rangeRt]),
     body: t.type({ serviceNames: jsonRt.pipe(t.array(t.string)) }),
   }),
   options: { tags: ['access:apm'] },
   handler: async (resources) => {
-    const {
-      context,
-      params,
-      request,
-      plugins: { security, logsDataAccess },
-    } = resources;
+    const { context, params, request } = resources;
+    const coreContext = await context.core;
 
-    const [coreContext, logsDataAccessStart] = await Promise.all([
-      context.core,
-      logsDataAccess.start(),
-    ]);
+    const entitiesESClient = await createEntitiesESClient({
+      request,
+      esClient: coreContext.elasticsearch.client.asCurrentUser,
+    });
 
-    const {
-      environment,
-      kuery,
-      offset,
-      start,
-      end,
-      probability,
-      documentType,
-      rollupInterval,
-      bucketSizeInSeconds,
-    } = params.query;
+    const { environment, start, end } = params.query;
 
     const { serviceNames } = params.body;
 
     if (!serviceNames.length) {
```

Comment:
Not part of this PR:

Should there also be a validation on how many service names this endpoint can receive to prevent it from being exploited?

Reply:
Indeed, we could add the validation; however, as with other internal APIs, we haven't included it. Please feel free to create a ticket to address this.